api/control/v1alpha1/types.go

```diff
@@ -20,6 +20,34 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// TemplateExpansionAnnotationKey, when paired with the value "true" in an annotation of
+// a workload object in a WDS, indicates that Go template expansion should be
+// bundled with propagation from core to WEC.
+//
+// Go template expansion means to (1) parse each leaf string of the object as a Go template
+// as defined in the Go standard package "text/template", and (2) for each WEC, replace that
+// leaf string with the string that results from expanding this template
+// (`Template.Execute`) using data about the WEC. This data is a map from parameter
+// name to parameter value.
+//
+// The parameters for a given WEC are collected from two sources.
+// One source is a ConfigMap object, if it exists, that: (a) has the same name as the WEC's
+// inventory object, (b) is in the namespace named "customization-parameters", and (c) is
+// in the Inventory and Transport Space (ITS). In particular, the string and binary data entries
+// whose name is valid as a Go language identifier provide parameters.
+// Also, if not defined that way, there is a parameter whose name is "clusterName" and
+// whose value is the name of the WEC's inventory object.
+//
+// Note that this sort of customization has limited applicability.  It can only be used where
+// the un-expanded string passes the validation conditions of the relevant object type.
+// For more broadly applicable customization, see Customizer objects.
+
+const TemplateExpansionAnnotationKey string = "control.kubestellar.io/expand-templates"
+
+// ParameterConfigMapNamespace is the namespace in the ITS that holds ConfigMap objects that provide
+// parameter values for template expansion.
+const ParameterConfigMapNamespace = "customization-parameters"
```

Comment:
I didn't understand this definition. 
do you plan to use a single namespace to hold customization configmaps for all WECs? why?
if user A has permissions to customize WEC1 and user B has permissions to customize WEC2 - now both users can access both maps.
why aren't you using the WEC namespace?
each WEC namespace can hold inside it's customization configmap...

Reply:
The mailbox namespaces are a private implementation detail, not part of the user interface.